CanvasColorSpaceProposal.md

```diff
@@ -140,18 +140,23 @@ IDL
 <pre>
 typedef (Uint8ClampedArray or Uint16ClampedArray or Float32Array) ImageDataArray;
 
-[Constructor(unsigned long sw, unsigned long sh, optional CanvasColorSpace colorSpace = "legacy-srgb"),
- Constructor(ImageDataArray data, unsigned long sw, optional unsigned long sh, optional CanvasColorSpace colorSpace),
+dictionary ImageDataColorSettings {
+  CanvasColorSpace colorSpace = "srgb";
+  CanvasPixelFormat pixelFormat = "8-8-8-8";
```

Comment:
I think we need to keep ImageDataColorSettings similar to CanvasRenderingContext2DSettings because PutImageData needs to know the pixel format of the image data it wants to color convert to the color space of the destination canvas. If we create an ImageData with, for example, p3 and Uint16ClampedArray, how PutImageData knows if the p3 is in 10-10-10-2 pixel format or 12-12-12-12 pixel format? I also believe we should clarify which type of ImageDataArray must be used for 10-10-10-2 and 12-12-12-12 pixel formats.